data/src/main/java/com/linkedin/data/codec/DataCodec.java

```diff
@@ -54,6 +55,34 @@
    */
   byte[] listToBytes(DataList list) throws IOException;
 
+  /**
+   * Serialize a {@link DataMap} to a {@link ByteString}.
+   *
+   * @param map to serialize.
+   * @return the output serialized from the {@link DataMap}.
+   * @throws IOException if there is a serialization error.
+   */
+  default ByteString mapToByteString(DataMap map) throws IOException
```

Comment:
Is it somewhat more visible if we put this two util functions to `DataMapUtils.java`

Reply:
We already have equivalents of these in DataMapUtils. The entire point behind introducing them here and invoking these versions from DataMapUtils is to optimize memory copies, instead of just wrapping a ByteString around a merged byte array.

Before this change, memory copies are 3x:

DataMap -> Codec Buffer -> FastByteArrayOutputStream internal buffer -> join bytes

After this change it becomes 2x for JSON:

DataMap -> Codec Buffer -> FastByteArrayOutputStream internal buffer

And 1x for Protobuf:

DataMap -> FastByteArrayOutputStream internal buffer

Reply:
Well, for understanding it more, 
can you tell me where the `join bytes` is saved
and where the `Codec Buffer ` is saved